### PR TITLE
Fixes processing of deleted objects in SSair

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/Monkestation/MonkeStation/pull/686
- https://github.com/tgstation/tgstation/pull/53735
- https://github.com/shiptest-ss13/Shiptest/pull/585

Original pr from monke, auxmos diff from shiptest, took the pr from monke

## About The Pull Request

Fixes runtimes with atmos machines process

> qdel'd objects could still process in SSair because SSair utilises a cache whenever it has to resume processing runs from a partially completed state from running out of processing time.
> Of all the things that processed on SSair, only one thing actually took care to remove itself from the cache as well on deletion.
> This is an important subsystem and the processing lists should not be public. Objects don't need to know how SSair works, they just call the proc to add when they start processing and call the proc to remove when they finish.

## Why It's Good For The Game
Shouldn't have any functional difference in game. Should reduce runtimes, especially with portable atmospherics objects

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

testers

</details>

## Changelog
:cl: RKz, timberpoes, Ninjanomnom, CoffeeKat, DoctorSquishy
fix: Fixed runtimes in the air subsystem, where deleted objects were still processed.
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
This could additionally be relevant to #8706, as that PR also involves machine processing

File: code/controllers/subsystem/air.dm [hunk 1]
```diff
@@ -380,15 +380,15 @@ SUBSYSTEM_DEF(air)
 	//cache for sanic speed (lists are references anyways)
 	var/list/currentrun = src.currentrun
 	while(currentrun.len)
-		var/obj/machinery/M = currentrun[currentrun.len]
+		var/obj/machinery/Machinery = currentrun[currentrun.len]
```
Comment (itsmeow) [inline code/controllers/subsystem/air.dm]: capitalized var names are bad

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/air.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/controllers/subsystem/air.dm]: you're bad 😢 

fixed in 25a1580

File: code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm [hunk 2]
```diff
@@ -31,6 +31,7 @@
 /obj/machinery/atmospherics/components/unary/thermomachine/Initialize(mapload)
 	. = ..()
 	initialize_directions = dir
+	SSair.start_processing_machine(src) //initialize thermomachines to SSairs list of tickable machines
```
Comment (itsmeow) [inline code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm]: does this ever stop processing?

Comment (itsmeow) [reply code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm]: isn't this already enabled in `/obj/machinery/atmospherics`?

Comment (Tsar-Salat) [reply code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm]: Yeah. Fixed in 1629067

**Review by PowerfulBacon — APPROVED** [review]

**Review by itsmeow — APPROVED** [review]

**Pull request merged**